### PR TITLE
bpo-41177: Add iterator support to ConvertingList, ConvertingTuple, and items() to ConvertingDict

### PR DESCRIPTION
**Pull request by godotgildor**
Add iterator support for logging.config.ConvertingList and logging.config.ConvertingTuple
Add items() support for logging.config.ConvertingDict.

https://bugs.python.org/issue41177

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@godotgildor

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Review by vsajip — CHANGES_REQUESTED** [review]
I think tests should be added, but the NEWS file removed (as the APIs being changed are internal, they don't need a NEWS entry - I'll add the `skip-news` label so it doesn't cause a block to merging).

File: Lib/logging/config.py [hunk 1]
```diff
@@ -350,6 +358,10 @@ def __getitem__(self, key):
         # Can't replace a tuple entry.
         return self.convert_with_key(key, value, replace=False)
 
+    def __iter__(self):
```
Comment (serhiy-storchaka) [inline Lib/logging/config.py:361]: Is it needed? Are any tests failed without it?

Comment (vsajip) [reply Lib/logging/config.py:361]: No tests fail without it, but the request is for additional functionality (which would need its own tests, of course). The issue gives more info about the use case.

Comment (serhiy-storchaka) [reply Lib/logging/config.py:361]: I suppose that tests for iterating ConvertingList and ConvertingTuple would expose that there is no need to add `__iter__()`.

Tests are necessary.

**Comment by vsajip** [comment]
I requested that tests be added, but the OP has not responded to that. The requested functionality could be added (the use case in the issue is not unreasonable) but it's not an especial priority - the OP can implement tests if they want to hurry things along, and I'll review it.

**Comment by serhiy-storchaka** [comment]
The original issue has been closed.

**Pull request closed without merging**